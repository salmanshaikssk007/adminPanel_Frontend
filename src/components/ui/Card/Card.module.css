.card {
  box-shadow: 0 4px 8px 0 #00000033;
  transition: 0.3s;
  border-radius: 5px;
  padding: 10px;
  margin: 5px 10px;
}
.card:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
  border-radius: 5px 5px 0 0;
}
.container {
  padding: 10px;
}
