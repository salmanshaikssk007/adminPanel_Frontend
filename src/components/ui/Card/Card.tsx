import React from 'react';
import Styles from './Card.module.css';

interface props {
  children: React.ReactNode;
}

const Card = (props: props) => {
  return (
    <div data-testid="card_ui" className={Styles.card}>
      <div data-testid="card_content_ui" className={Styles.container}>
        {props.children}
      </div>
    </div>
  );
};

export default Card;
