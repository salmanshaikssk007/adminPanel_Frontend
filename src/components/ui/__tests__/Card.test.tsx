import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Card from '../Card/Card';

configure({ adapter: new Adapter() });

describe('<Card />', () => {
  test('testing children', () => {
    const wrapper = shallow(<Card>children</Card>);
    expect(wrapper.text()).toBe('children');

    expect(wrapper).toMatchSnapshot();
  });
});
