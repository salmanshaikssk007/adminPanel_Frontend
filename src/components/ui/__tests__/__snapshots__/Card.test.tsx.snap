// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> testing children 1`] = `
<div
  className="card"
  data-testid="card_ui"
>
  <div
    className="container"
    data-testid="card_content_ui"
  >
    children
  </div>
</div>
`;
